### PR TITLE
1850/remove padding horizon

### LINKED ISSUE #1734: Bound needs to accept datetime obects.
If you try to use bounds in a datetime axis, it throws an error because bounds accept only a tuple of floats.

### LINKED ISSUE #1850: remove gaps on left and right sides of horizon chart
There is some gap on each sides of the plot that should be filled

### PR DESCRIPTION
**Pull request by matdrapeau**
issues: closes #1850 closes #1734

### CODE REVIEW

File: bokeh/models/axes.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Axis(GuideRenderer):
     Where should labels and ticks be located in relation to the axis rule.
     """)
 
-    bounds = Either(Auto, Tuple(Float, Float), help="""
+    bounds = Either(Auto, Tuple(Float, Float), Tuple(Datetime, Datetime), help="""
```
Comment (bryevdv) [inline bokeh/models/axes.py]: Do we know if this works for sure? I feel like I tried this and it was not sufficient but I may be misremembering.

Comment (damianavila) [inline bokeh/models/axes.py]: Well, using this:
```python
import numpy as np

from bokeh.sampledata.stocks import AAPL, FB, GOOG, IBM, MSFT
from bokeh.plotting import *

output_file("stocks.html", title="stocks.py example")

left_range = np.datetime64('2000-03-01')
right_range = np.datetime64('2006-03-01')

p1 = figure(x_axis_type = "datetime")

p1.line(np.array(AAPL['date'], 'M64'), AAPL['adj_close'], color='#A6CEE3', legend='AAPL')
p1.line(np.array(FB['date'], 'M64'), FB['adj_close'], color='#1F78B4', legend='FB')
p1.line(np.array(GOOG['date'], 'M64'), GOOG['adj_close'], color='#B2DF8A', legend='GOOG')
p1.line(np.array(IBM['date'], 'M64'), IBM['adj_close'], color='#33A02C', legend='IBM')
p1.line(np.array(MSFT['date'], 'M64'), MSFT['adj_close'], color='#FB9A99', legend='MSFT')

p1.title = "Stock Closing Prices"
p1.grid.grid_line_alpha=0.3
p1.xaxis.axis_label = 'Date'
p1.yaxis.axis_label = 'Price'
p1.xaxis.bounds = (left_range, right_range)

show(p1)  # open a browser
```

I get this plot...

bounds

Seems to work...

**Comment by bryevdv** [comment]
sweet! LGTM

**Comment by damianavila** [comment]
ok, code LGTM too... merging now...

**Pull request merged**